kore/src/Kore/Reachability/Claim.hs

```diff
@@ -336,24 +332,29 @@ transitionRule claims axiomGroups = transitionRuleWorker
               | otherwise -> pure (Claimed a)
       | otherwise = pure claimState
 
-    -- TODO (virgil): Wrap the results in GoalRemainder/GoalRewritten here.
-    --
-    -- thomas.tuegel: "Here" is in ApplyClaims and ApplyAxioms.
-    --
-    -- Note that in most transitions it is obvious what is being transformed
-    -- into what, e.g. that a `ResetGoal` transition transforms
-    -- `GoalRewritten` into `Goal`. However, here we're taking a `Goal`
-    -- and transforming it into `GoalRewritten` and `GoalRemainder` in an
-    -- opaque way. I think that there's no good reason for wrapping the
-    -- results in `derivePar` as opposed to here.
-
-    transitionRuleWorker ApplyClaims (Claimed claim) =
-        applyClaims claims claim
+    transitionRuleWorker ApplyClaims (Claimed claim) = do
+        applied <- applyClaims claims claim
+        case applied of
+            -- If the result is \\bottom, the claim is proven on the
+            -- current branch.
+            ReachedBottom -> return Proven
```

Comment:
This is already handled in `checkImplication`. Why do we need to handle it here, too?